extensions/wikia/PageHeader/ActionButton.class.php

```diff
@@ -0,0 +1,145 @@
+
+		return $this->buttonAction;
+	}
+
+	public function getDropdownActions(): array {
+		// items to be added to "edit" dropdown
+		$actions = [
+			'edit',
+			've-edit',
+			'formedit',
+			'history',
+			'move',
+			'protect',
+			'unprotect',
+			'delete',
+			'undelete',
+			'replace-file',
+			'talk',
+			'edit-mobile-main-page'
+		];
+
+		// Enable to modify actions list on dropdown
+		wfRunHooks( 'PageHeaderDropdownActions', [ &$actions ] );
+
+		return array_map( function( $action ) {
+			$ret = $this->contentActions[$action];
+			if ( isset( $ret['id'] ) ) {
+				$ret['data-tracking'] = $ret['id'] . '-dropdown';
+			}
+
+			return $ret;
+		}, array_filter( $actions, function( $action ) {
+			return isset( $this->contentActions[$action] );
+		}));
+	}
+
+	private function prepareActionButton() {
+		wfRunHooks( 'BeforePrepareActionButtons', [ $this, &$this->contentActions ] );
+
+		$isDiff = !is_null( $this->request->getVal( 'diff' ) );
+
+		// "Add topic" action
+		if ( isset( $this->contentActions['addsection'] ) ) {
+			// remove on diff pages (RT #72666)
+			if ( $isDiff ) {
+				unset( $this->contentActions['addsection'] );
+			}
+		}
+
+		// handle protected pages (they should have viewsource link and lock icon) - BugId:9494
+		if ( isset( $this->contentActions['viewsource'] ) &&
+			!$this->title->isProtected() &&
+			!$this->title->isNamespaceProtected( $this->user ) &&
+			!$this->user->isLoggedIn() /* VOLDEV-74: logged in users should see the viewsource button, not edit */
+		) {
+			// force login to edit page that is not protected
+			$this->contentActions['edit'] = $this->contentActions['viewsource'];
+			$this->contentActions['edit']['text'] = wfMessage( 'page-header-action-button-edit' )->escaped();
+			$this->buttonIcon = self::LOCK_ICON;
+			unset( $this->contentActions['viewsource'] );
+		}
+
+		// If cascade protected, show viewsource button - BugId:VE-89
+		if ( isset( $this->contentActions['edit'] ) && $this->title->isCascadeProtected() ) {
+			$this->contentActions['viewsource'] = $this->contentActions['edit'];
+			$this->contentActions['viewsource']['text'] =
+				wfMessage( 'page-header-action-button-viewsource' )->escaped();
+			$this->buttonIcon = self::LOCK_ICON;
+			unset( $this->contentActions['edit'] );
+		}
+
+		// "Add topic"
+		if ( isset( $this->contentActions['addsection'] ) ) {
+			$action = $this->contentActions['addsection'];
+			$action['text'] = wfMessage( 'page-header-action-button-add-topic' )->escaped();
+			$this->buttonAction = $action;
+		}  // ve-edit
+		else if ( isset( $this->contentActions['ve-edit'] ) && $this->contentActions['ve-edit']['main'] ) {
+			$this->buttonAction = $this->contentActions['ve-edit'];
+			unset( $this->contentActions['ve-edit'] );
+		} // edit
+		else if ( isset( $this->contentActions['edit'] ) ) {
+			$this->buttonAction = $this->contentActions['edit'];
+			unset( $this->contentActions['edit'] );
+		} // view source
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)